crates/agentgateway/src/http/auth.rs

```diff
@@ -77,7 +86,7 @@ pub async fn apply_late_backend_auth(
 		BackendAuth::Passthrough {} => {},
 		BackendAuth::Key(k) => {},
 		BackendAuth::Gcp {} => {},
-		BackendAuth::Aws {} => {
+		BackendAuth::Aws { .. } => {
```

Comment:
maybe we want to make Aws an enum of ExplictConfig (what you added) or Implicit (what we have before). (better names welcome)